src/Symfony/Bundle/FrameworkBundle/CacheWarmer/TranslationsCacheWarmer.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\CacheWarmer;
+
+use Symfony\Component\HttpKernel\CacheWarmer\CacheWarmerInterface;
+use Symfony\Component\HttpKernel\CacheWarmer\WarmableInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Generates the catalogues for translations.
+ *
+ * @author Xavier Leune <xavier.leune@gmail.com>
+ */
+class TranslationsCacheWarmer implements CacheWarmerInterface
+{
+    private $translator;
+
+    /**
+     *
+     * @param TranslatorInterface $translator A Translator instance
+     */
+    public function __construct(TranslatorInterface $translator)
+    {
+        $this->translator = $translator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function warmUp($cacheDir)
+    {
+        if ($this->translator instanceof WarmableInterface) {
```

Comment:
We cannot. Otherwise we forbid decorating the translator with a non-warmable one (or we need to add complex logic in a compiler pass to determine whether we should disable this cache warmer entirely and remove it from the container to avoid being broken)